src/cache/precomputed_quantities.jl

```diff
@@ -399,25 +373,84 @@ function thermo_state(
     return get_ts(ρ, p, θ, e_int, q_tot, q_pt)
 end
 
-function thermo_vars(moisture_model, microphysics_model, ᶜY, K, Φ)
-    energy_var = (; e_int = specific(ᶜY.ρe_tot, ᶜY.ρ) - K - Φ)
+function ᶜthermo_state(
+    thermo_params;
+    ρ = nothing,
+    p = nothing,
+    θ = nothing,
+    e_int = nothing,
+    q_tot = nothing,
+    q_pt = nothing,
+)
+
+    get_ts(ρ, ::Nothing, θ, ::Nothing, ::Nothing, ::Nothing) =
+        TD.PhaseDry_ρθ(thermo_params, ρ, θ)
+    get_ts(ρ, ::Nothing, θ, ::Nothing, q_tot, ::Nothing) =
+        TD.PhaseEquil_ρθq(thermo_params, ρ, θ, q_tot)
+    get_ts(ρ, ::Nothing, θ, ::Nothing, ::Nothing, q_pt) =
+        TD.PhaseNonEquil_ρθq(thermo_params, ρ, θ, q_pt)
+    get_ts(ρ, ::Nothing, ::Nothing, e_int, ::Nothing, ::Nothing) =
+        TD.PhaseDry_ρe(thermo_params, ρ, e_int)
+    get_ts(ρ, ::Nothing, ::Nothing, e_int, q_tot, ::Nothing) =
+        TD.PhaseEquil_ρeq(
+            thermo_params,
+            ρ,
+            e_int,
+            q_tot,
+            3,
+            eltype(thermo_params)(0.003),
+        )
+    get_ts(ρ, ::Nothing, ::Nothing, e_int, ::Nothing, q_pt) =
+        TD.PhaseNonEquil(thermo_params, e_int, ρ, q_pt)
+    get_ts(::Nothing, p, θ, ::Nothing, ::Nothing, ::Nothing) =
+        TD.PhaseDry_pθ(thermo_params, p, θ)
+    get_ts(::Nothing, p, θ, ::Nothing, q_tot, ::Nothing) =
+        TD.PhaseEquil_pθq(thermo_params, p, θ, q_tot)
+    get_ts(::Nothing, p, θ, ::Nothing, ::Nothing, q_pt) =
+        TD.PhaseNonEquil_pθq(thermo_params, p, θ, q_pt)
+    get_ts(::Nothing, p, ::Nothing, e_int, ::Nothing, ::Nothing) =
+        TD.PhaseDry_pe(thermo_params, p, e_int)
+    get_ts(::Nothing, p, ::Nothing, e_int, q_tot, ::Nothing) =
+        TD.PhaseEquil_peq(thermo_params, p, e_int, q_tot)
+    get_ts(::Nothing, p, ::Nothing, e_int, ::Nothing, q_pt) =
+        TD.PhaseNonEquil_peq(thermo_params, p, e_int, q_pt)
+
+    return @. lazy(get_ts(ρ, p, θ, e_int, q_tot, q_pt))
+end
+
+function thermo_vars(moisture_model, microphysics_model, Y_c, K, Φ)
```

Comment:
Is there a reason we're using `Y_c`, `ᶜY`, `Yc` and `Y.c` to denote the prognostic center valued variables in this PR? Is there a specific convention you'd like us to follow with this lazy broadcast change?

Reply:
In general, I think it should be Y for the entire state (since it contains both center and face fields internally), `ᶜvar` for center fields and `ᶜfunc` for functions that return lazy broadcasted objects or fields. Some of the inconsistencies you mention follow existing patterns locally, but I can make it self-consistent in the part of the code this PR touches.